src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectMethodExecutor.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    /// <summary>
+    /// This class represents a general purpose executor for a method(static/instance) on an object using MethodInfo.
+    /// This executor is an alternative approach for MethodInfo.Invoke(). 
+    /// Given the MethodInfo and the type of the target object, it creates a delegate and calls the method using it.
+    /// The delegate is created dynamically using the Linq expressions. 
+    /// The executor should be cached to get the maximum performance benefit or else compiling Linq expression adds overhead.
+    /// </summary>
+    public class ObjectMethodExecutor
+    {
+        private ActionExecutor _executor;
+
+        public ObjectMethodExecutor(MethodInfo methodInfo, TypeInfo targetTypeInfo)
```

Comment:
My preference would be the opposite. Make the constructor private, and make the static factory public. The expectation is that a constructor won't do significant work.